st2api/st2api/controllers/v1/triggers.py

```diff
@@ -315,13 +318,64 @@ def __get_by_name(trigger_name):
             return []
 
 
-class TriggerInstanceController(resource.ResourceController):
+class TriggerInstanceMixin(RestController):
+    model = TriggerInstanceAPI
+    access = TriggerInstance
+
+
+class TriggerInstanceResendController(TriggerInstanceMixin, resource.ResourceController):
+    supported_filters = {}
+
+    def __init__(self, *args, **kwargs):
+        super(TriggerInstanceResendController, self).__init__(*args, **kwargs)
+        self.trigger_dispatcher = TriggerDispatcher(LOG)
+
+    class TriggerInstancePayload(object):
+        def __init__(self, payload=None):
+            self.payload = payload or {}
+
+        def validate(self):
+            if self.payload:
+                assert isinstance(self.payload, dict)
+
+            return True
+
+    @jsexpose(status_code=http_client.OK)
+    def post(self, trigger_instance_id):
+        """
+        Re-send the provided trigger instance optionally specifying override parameters.
+
+        Handles requests:
+
+            POST /triggerinstance/<id>/re_emit
+            POST /triggerinstance/<id>/re_send
```

Comment:
Not really a fan of supporting 2 different endpoints for the same action, especially since this is a new endpoint and we don't need to do that to maintain backward compatibility or other similar reason.

`There should be one-- and preferably only one --obvious way to do it.`